api/routers/v1/timelog.py

```diff
@@ -143,25 +145,95 @@ async def add_task(task: TaskNewSchema, current_user=Depends(get_current_user),
     return result
 
 
-def validate_task(task: TaskSchema, db: Session):
+@router.put("/tasks/{task_id}", response_model=TaskSchema)
+async def update_task(
+    task_id: int, task: TaskUpdateSchema, current_user=Depends(get_current_user), db: Session = Depends(get_db)
+):
+    """
+    Update an task with any of the following data:
+
+    - **user id**: the user id
+    - **project_id**: the project the task is associated with
+    - **story**: the task story
+    - **description**: the task description
+    - **task type**: the task type
+    - **date**: the task date (YYYY-MM-DD)
+    - **start time**: the task start time in 24h time notation (HH:mm)
+    - **end time**: the task end time in 24h time notation (HH:mm)
+
+    \f
+    :param item: User input.
+    """
+    existing_task = TaskService(db).get_task(task_id)
+    if not existing_task:
+        raise HTTPException(status_code=404, detail=f"Task with id {task_id} not found")
+    user_id_to_check = task.user_id if task.user_id else existing_task.user_id
+    if current_user.id != user_id_to_check:
+        raise HTTPException(status_code=403, detail="You are not authorized to update tasks for this user.")
+    if not task.user_id:
+        task.user_id = user_id_to_check
```

Comment:
This check is been done twice, this is minor but can simplify a bit with a reorder:

```suggestion
    if not task.user_id:
        task.user_id = existing_task.user_id
    if current_user.id != task.user_id:
        raise HTTPException(status_code=403, detail="You are not authorized to update tasks for this user.")
```

Reply:
Since I changed the update to simply merge the changes with the existing task, this is probably no longer relevant.